arbnode/batch_poster.go

```diff
@@ -1232,7 +1238,21 @@ func (b *BatchPoster) maybePostSequencerBatch(ctx context.Context) (bool, error)
 		b.building.msgCount++
 	}
 
-	if !forcePostBatch || !b.building.haveUsefulMessage {
+	var disablePosting bool
+	batchDuration := time.Since(firstMsgTime)
+	if (config.ReorgResistanceMargin > 0 && config.ReorgResistanceMargin < config.MaxDelay) &&
+		(batchDuration >= config.MaxDelay-config.ReorgResistanceMargin) &&
+		(batchDuration <= config.MaxDelay || (batchDuration <= config.MaxDelay+config.ReorgResistanceMargin && !config.IgnoreReorgResistanceMargin)) {
```

Comment:
I'm personally having trouble reasoning through this compound condition. For example, I wonder why we are only checking `config.IgnoreReorgResistanceMargin` in the last parenthetical `||` instead of in the other places where the value of the `config.ReorgResistanceMargin` is being checked. Like, why don't we only ignore the `config.ReorgResistanceMargin` when checking `<=` instead of for both `<=` and `>=`?

Maybe some test-cases would make it easier to see the truth table of when this clause should disable posting?

Reply:
I agree with @eljobe on this one.